### PR TITLE
Register bloc

### PR DESCRIPTION
**Pull request by HassanSobhy**

### CODE REVIEW

File: lib/models/register/register_response_model.dart [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+class RegisterResponseModel{
+  bool status;
+  String message;
+  RegisterUserDataModel registerUserDataModel;
+
+  RegisterResponseModel.fromJson(Map<String, dynamic> json)
+  {
+    status = json['status'];
+    message = json['message'];
+    registerUserDataModel = json['data'] != null ? RegisterUserDataModel.fromJson(json['data']) : null;
```
Comment (AmrMahm0ud) [inline lib/models/register/register_response_model.dart]: format file

File: lib/models/register/register_user_data_model.dart [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class RegisterUserDataModel{
```
Comment (AmrMahm0ud) [inline lib/models/register/register_user_data_model.dart]: format file

File: lib/modules/auth/register/bloc/register_bloc.dart [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import 'dart:async';
```
Comment (AmrMahm0ud) [inline lib/modules/auth/register/bloc/register_bloc.dart:1]: format file

File: lib/modules/auth/register/bloc/register_bloc.dart [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
```
Comment (AmrMahm0ud) [inline lib/modules/auth/register/bloc/register_bloc.dart:2]: make validation inside bloc

File: lib/modules/auth/register/bloc/register_bloc.dart [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+import 'package:bloc/bloc.dart';
```
Comment (AmrMahm0ud) [inline lib/modules/auth/register/bloc/register_bloc.dart:3]: to get the validator class to your branch merge register-bloc development 
so
on register-bloc ==> git pull 
checkout to development ==> git pull
checkout to register-bloc ==> git pull
after that merge using 
git merge register-bloc development

**Review by AmrMahm0ud — COMMENTED** [review]
format all files and check comments

**Comment by AmrMahm0ud** [comment]
solve conflicts

File: lib/models/register/register_response_model.dart (hunk 1 above)
Comment (HassanSobhy) [reply lib/models/register/register_response_model.dart]: done

File: lib/models/register/register_user_data_model.dart (hunk 2 above)
Comment (HassanSobhy) [reply lib/models/register/register_user_data_model.dart]: done

File: lib/modules/auth/register/bloc/register_bloc.dart (hunk 3 above)
Comment (HassanSobhy) [reply lib/modules/auth/register/bloc/register_bloc.dart:1]: done

**Review by AmrMahm0ud — APPROVED** [review]
Good Job

**Pull request merged**